Key Vault 2025-07-01 API version

melissamserv opened the pull request:
Moving from ‘2025-06-01-preview’ to ‘2025-07-01’ stable, not adding any additional changes.

mikeharder commented:
`Approved-Avocado` due to this bug (which should be fixed before your next PR): https://github.com/Azure/avocado/issues/159

melissamserv commented:
/azp run SDK Validation - Go

mikekistler reviewed (changes requested):
Azure versioning policy does not allow a preview and GA API version to have the same date component. Since the prior preview version is 2025-06-01-preview, the API version for this GA must be 2025-06-02 or later.

raych1 commented on specification/keyvault/Security.KeyVault.Settings/tspconfig.yaml:
```diff
@@ -27,6 +27,7 @@ options:
   "@azure-tools/typespec-go":
     service-dir: "sdk/security/keyvault"
     package-dir: "azadmin/settings"
```
`package-dir` was deprecated and the value is appended to `emitter-output-dir`.

gracewilcox reviewed (commented):
Adding on to Ray's comments, there's been some updates for how the tspconfig.yaml file is used. This should be the correct configuration for Go.

melissamserv committed:
Copy files from preview/2025-06-01-preview to stable/2025-07-01.

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous spec.

melissamserv committed:
Update API version to 2025-07-01 in stable.

Updated the API version from 2025-06-01-preview to 2025-07-01.

melissamserv committed:
Add package-2025-07-01 tag in readme.

melissamserv committed:
Copy TSP examples from 2025-06-01-preview to 2025-07-01.

Copied the files in a separate commit.
This allows reviewers to easily diff subsequent changes against the previous version.

melissamserv committed:
Update TSP examples version to 2025-07-01.

Updated the API version from 2025-06-01-preview to 2025-07-01.

melissamserv committed:
Add 2025-07-01 API version enum value to TSP.

melissamserv committed:
Update ReleasePolicy-example format for TypeSpec validation compliance.

specification/keyvault/Security.KeyVault.Keys/examples/2025-07-01/ReleasePolicy-example.json:1:1 - error @azure-tools/typespec-autorest/example-loading: Skipping example file ReleasePolicy-example.json because it does not contain an operationId and/or title.

melissamserv committed:
Update config for go.

mikekistler reviewed (approved):
Looks good! 👍 

Thanks for fixing the api-version.

mikeharder committed:
Merge branch 'main' into keyvault-Microsoft.KeyVault-2025-06-01

mikeharder committed:
empty

raych1 commented:
/azp run SDK Validation - .NET

mikeharder committed:
Merge branch 'main' into keyvault-Microsoft.KeyVault-2025-06-01

chlowell committed:
Merge branch 'main' into keyvault-Microsoft.KeyVault-2025-06-01
